compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -698,14 +738,20 @@ def IREELinalgExt_OnlineAttentionOp : IREELinalgExt_PureOp<"online_attention",
     AffineMap getValueMap() {
       return getIndexingMapsArray()[2];
     }
+    std::optional<AffineMap> getMaskMap() {
+      if (getMask()) {
+        return cast<AffineMap>(getIndexingMapsArray()[4]);
+      }
+      return std::nullopt;
+    }
     AffineMap getOutputMap() {
-      return getIndexingMapsArray()[3];
+      return cast<AffineMap>(getIndexingMapsArray()[getMask() ? 5 : 4]);
     }
     AffineMap getMaxMap() {
-      return getIndexingMapsArray()[4];
+      return cast<AffineMap>(getIndexingMapsArray()[getMask() ? 6 : 5]);
     }
     AffineMap getSumMap() {
-      return getIndexingMapsArray()[5];
+      return cast<AffineMap>(getIndexingMapsArray()[getMask() ? 7 : 6]);
     }
```

Comment:
I guess we can try move it in the TD as the final arg, while keeping the assembler as is S.T it still visually looks like a dps input.

Reply:
Cleaned it up to refer to `getNumDpsInputs()` instead. Having to put mask at the end of Affine Map would require some awkward and troublesome "gymnastics" (Stanley's favorite word) to work with the MLIR

Reply:
We'd want mask's map to live right after scale's map since it should technically be considered an dpsInput as opposed to a dpsInit. 

But since we are printing the indexing maps as is (without specifying the order in the assembler), if we want the mask to show up in the IR after scale, we kind of need to preserve this ordering. (or update the assembler to hard code the printing of maps) 
 
I think it's clean enough for now, alternatively we can hardcode the assembler to show mask operand and the map at the place we'd want it, while in reality it all lives at the end of the vector. LMK what y'all think :)